src/components/auth/registerUser.js

```diff
@@ -0,0 +1,228 @@
+import React, { Component } from "react";
+import Input from "../common/input";
+import Button from "../common/button";
+import "./../../assets/authAssets/test.css";
+import "bootstrap/dist/css/bootstrap.min.css";
+import { connect } from "react-redux";
+
+import { signUpAction } from "../../actions/authActions/signUpAction";
+
+export class RegisterUser extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newUser: {
+        username: "",
+        email: "",
+        password: "",
+        passwordConfirm: ""
+      },
+      isValidated: false,
+      errors:{}
+    };
+    this.handleFormSubmit = this.handleFormSubmit.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+  }
+
+  handleInputChange(event) {
+    const target = event.target;
+    const value = target.type === "checkbox" ? target.checked : target.value;
+    const name = target.name;
+    // event.target.classList.add("active");
+    // this.showInputError(event.target.name);
+    event.target.classList.add("active");
+    this.showInputError(event.target.name);
+
+    this.setState(
+      prevState => {
+        return {
+          newUser: {
+            ...prevState.newUser,
+            [name]: value
+          }
+        };
+      },
+      () => console.log({ user: this.state.newUser })
+    );
+  }
+
+  showInputError(refName) {
+    const validity = document.getElementById(`${refName}`).checkValidity;
+    const label = document.getElementById(`${refName}`).name;
+    const error = document.getElementById(`${refName}Error`);
+    const isPassword = refName.indexOf("password") !== -1;
+    const isPasswordConfirm = refName === "passwordConfirm";
+
+    // debugger;
+    if (isPasswordConfirm) {
+      if (
+        document.getElementById("password").value !==
+        document.getElementById("passwordConfirm").value
```

Comment:
Use event.target.value to pick data from instead of document.getElementById

Reply:
I have removed this type of validation